### ISSUE TITLE
cohttp-eio(server): reject request with 411 if Content-Length is not provided.

### ISSUE THREAD

**Issue by bikallem** [mirage/ocaml-cohttp#931]
According to RFC 7230
> A server MAY reject a request that contains a message body but not a Content-Length by responding with 411 > (Length Required).

Therefore reject requests if Content-Length is not provided but is required.

**Comment by mseri** [comment]
It says may. Should we leave it as an implementation detail for the library users to implement?

**Comment by bikallem** [comment]
> It says may. Should we leave it as an implementation detail for the library users to implement?

Yes, it will be opt in - implemented via middleware - but the functionality will be available from cohttp-eio package.
